compat/test/browser/suspense.test.js

```diff
@@ -2159,4 +2160,38 @@ describe('suspense', () => {
 			expect(scratch.innerHTML).to.equal('<div><p>hello world</p></div>');
 		});
 	});
+
+	it('should correctly render memod child component', () => {
```

Comment:
This test does not test the behaviour, it even passes on main. You might want to look at the reproduction to more closely reproduce the failure. Currently this component renders successfully and retains said state, what we want to assert is that it can attain new state